### DISCUSSION TITLE
Error response contract

### DISCUSSION THREAD

**Discussion by repko-artem** [BerriAI/litellm#4834] [Q&A] [answered]
Hi.
How do you handle error responses from LiteLLM proxy? We're using C# library https://github.com/betalgo/openai to make requests to OpenAI endpoint (can switch between embedded OpenAI, AzureOpenAI providers and LiteLLM using custom domain for requests). But with LiteLLM we have trouble that when error occurs (i.e. some bedrock violation for name of function) we get some stacktrace error instead of JSON with error that produces OpenAI or Azure OpenAI. So betalgo lib just can't recognize such answer as JSON and we can't get error message.

Is it possible to get errors from LiteLLM in structured JSON format like OpenAI providers do?

**Comment by ishaan-jaff** [comment]
> Is it possible to get errors from LiteLLM in structured JSON format like OpenAI providers do?

Hi @repko-artem - yes our goal is to have structured JSON exceptions. Could you share the stacktrace you're seeing. I'll fix it

**Comment by repko-artem** [comment]
@ishaan-jaff Hi.
Sorry, I was incorrect. LiteLLM responses with json, but it's incompatible with contract of the response of Betalgo/OpenAI lib:

Your response has error code of int type:
```
{
	"error": {
		"message": "Authentication Error, No connected db.",
		"type": "auth_error",
		"param": "None",
		"code": 401
	}
}
```
But Betalgo expects string code:
https://github.com/betalgo/openai/blob/fd3a4d923c18d3c27e029f72d785a1cb47c87055/OpenAI.SDK/ObjectModels/ResponseModels/BaseResponse.cs#L96

And we're getting exception:
```
System.Text.Json.JsonException: The JSON value could not be converted to System.String. Path: $.error.code | LineNumber: 0 | BytePositionInLine: 106.
```

How do you think who is right?) Should error code be string or int?
Asked Betalgo authors too: https://github.com/betalgo/openai/discussions/615

----
**Update**
----
If we look on official python OpenAI lib, the code should be a string:
https://github.com/openai/openai-python/blob/195c05a64d39c87b2dfdf1eca2d339597f1fce03/src/openai/types/shared/error_object.py#L11

**Reply by ishaan-jaff** [reply, marked as answer]
Started an issue for this @repko-artem, https://github.com/BerriAI/litellm/issues/4970 

Will push a fix today :)

**Reply by ishaan-jaff** [reply]
Fix pr here @repko-artem https://github.com/BerriAI/litellm/pull/4973
